### PR TITLE
Add decrypt option to decrypt body on demand

### PR DESCRIPTION
**Pull request by janbrummer**
Fixes: https://github.com/libkeepass/pykeepass/issues/359

### CODE REVIEW

**Comment by janbrummer** [comment]
@Evidlo Like so?

**Comment by Evidlo** [comment]
Looks good so far.  Just needs a test and a description added to the misc section of the README

**Comment by janbrummer** [comment]
So i tried to implement the necessary parts in Secrets and discovered that i was wrong with master_seed... it's actually the kdf algorithm salt. Sorry about that, somehow mixed the output during debugging keepassxc. Nevertheless the PR main part is still valid. I'll adjust the property and add the necessary requested bits.

File: pykeepass/kdbx_parsing/kdbx.py [hunk 1]
```diff
@@ -24,10 +24,12 @@ def check_signature(ctx):
             )
         )
     ),
-    "body" / Switch(
-        this.header.value.major_version,
-        {3: Body3,
-         4: Body4
-         }
+    If(this._.decrypt,
```
Comment (Evidlo) [inline pykeepass/kdbx_parsing/kdbx.py:27]: "body" should be outside `If`, I think.

File: pykeepass/pykeepass.py [hunk 2]
```diff
@@ -209,6 +212,11 @@ def transformed_key(self):
         and passed to `open` for faster database opening"""
         return self.kdbx.body.transformed_key
 
+    @property
+    def master_seed(self):
+       """bytes: master seed of database"""
```
Comment (A6GibKm) [inline pykeepass/pykeepass.py:217]: Is there more that can be said about this? e.g. is there an expected size?

File: pykeepass/kdbx_parsing/kdbx.py (hunk 1 above)
Comment (janbrummer) [reply pykeepass/kdbx_parsing/kdbx.py]: Code changed and this is obsolete.

File: pykeepass/pykeepass.py (hunk 2 above)
Comment (janbrummer) [reply pykeepass/pykeepass.py:220]: It's no longer the master_seed but the salt and if i remember correctly this is not fixed. Any idea?

**Comment by janbrummer** [comment]
Secrets works fine with this solution \o/ Ready for review.

File: pykeepass/pykeepass.py [hunk 3]
```diff
@@ -45,6 +45,7 @@ class PyKeePass():
             database is assumed to have no keyfile
         transformed_key (:obj:`bytes`, optional): precomputed transformed
             key.
+        decrypt (:obj:`bool`, optional): whether to decrypt payload or not.
```
Comment (A6GibKm) [inline pykeepass/pykeepass.py:48]: I think this needs more explanation, its not clear what it does from an outsider perspective.

File: pykeepass/pykeepass.py [hunk 4]
```diff
@@ -209,6 +215,12 @@ def transformed_key(self):
         and passed to `open` for faster database opening"""
         return self.kdbx.body.transformed_key
 
+    @property
+    def database_salt(self):
```
Comment (A6GibKm) [inline pykeepass/pykeepass.py:219]: Needs a test, if possible a small line explaining what it is too.

**Review by A6GibKm — COMMENTED** [review]
Thanks for doing this PR!

File: pykeepass/pykeepass.py (hunk 3 above)
Comment (janbrummer) [reply pykeepass/pykeepass.py:48]: Ok, then please help me out with a better description

File: pykeepass/pykeepass.py (hunk 4 above)
Comment (janbrummer) [reply pykeepass/pykeepass.py:219]: Ok, will add one.

File: pykeepass/pykeepass.py (hunk 3 above)
Comment (janbrummer) [reply pykeepass/pykeepass.py:48]: Ping :)

Comment (Evidlo) [reply pykeepass/pykeepass.py:48]: Maybe something like 

    whether to decrypt XML payload.  Set `False` to access outer header information without decrypting database.

Comment (janbrummer) [reply pykeepass/pykeepass.py:48]: Thanks for jumping in, i'll take it.

File: pykeepass/pykeepass.py (hunk 4 above)
Comment (janbrummer) [reply pykeepass/pykeepass.py:221]: Added.

**Pull request merged**